File: 백준/Silver/28278. 스택 2/스택 2.js
const input = require('fs').readFileSync(0, 'utf-8').toString().trim().split('\n');
const N = +input[0];
const stack = [];
let answer = '';

for (let i = 1; i <= N; i++) {
    const [action, x] = input[i].split(' ');
    
    if (action === '1') {
        stack.push(x);
    } else if (action === '2') {
        answer += `${stack.length > 0 ? stack.pop() : -1}\n`;
    } else if (action === '3') {
        answer += `${stack.length}\n`;
    } else if (action === '4') {
        answer += `${stack.length === 0 ? 1 : 0}\n`;
    } else if (action === '5') {
        answer += `${stack.length === 0 ? -1 : stack[stack.length - 1]}\n`;
    }
}

console.log(answer);

File: 백준/Silver/28278. 스택 2/스택 2.py
import sys

N = int(input())
commands = [[int(t) for t in line.split()] for line in sys.stdin]
stack = []

for cmd in commands:
    match cmd[0]:
        case 1:
            stack.append(cmd[1])
        case 2:
            print(stack.pop() if stack else -1)
        case 3:
            print(len(stack))
        case 4:
            print(0 if stack else 1)
        case 5:
            print(stack[-1] if stack else -1)
            